src/ngx_http_redis2_handler.c

```diff
@@ -266,6 +267,12 @@ ngx_http_redis2_process_header(ngx_http_request_t *r)
             return NGX_HTTP_UPSTREAM_INVALID_HEADER;
     }
 
+    clcf = ngx_http_get_module_loc_conf(r, ngx_http_core_module);
+    
+    if (!clcf->chunked_transfer_encoding) {
+        u->headers_in.content_length_n = b->last - b->pos;
```

Comment:
This looks wrong since the we may not have all the data inside the buffer (yet). This module does streaming processing instead of full buffering which is important for large redis value payload.

Reply:
Thank you for your review. It seems that this module already sent response headers in process_header callback function before getting content_length from redis reply. fmmm...